app/services/submit_to_app_store/prior_authority_payload_builder.rb

```diff
@@ -9,12 +9,20 @@ def payload
       { application_id: application.id,
         json_schema_version: 1,
         application_state: application.state,
-        application: data,
+        application: validated_data,
         application_type: 'crm4',
         application_risk: 'N/A',
         events: @include_events ? PriorAuthority::EventBuilder.call(application, data) : [] }
     end
 
+    def validated_data
+      built_data = data
```

Comment:
Should this logic not match https://github.com/ministryofjustice/laa-assess-crime-forms/pull/791/files#diff-d3a41f06a5d16530196db1064f66524b3318790b22adce1fd254a8c9f060c653 ?

Reply:
It won't exactly match it because this class already knows it's building a PA payload, and it actually needs to construct it from scratch so it needs to be memoised. On the other side the payload already exists, but on the flipside it could be PA or NSM.